internal/provider/resource_domain.go

```diff
@@ -0,0 +1,496 @@
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Represents a memory used metric for the domain",
+			},
+			"total_storage_capacity": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Represents a storage total metric for the domain",
+			},
+			"used_storage_capacity": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Represents a storage used metric for the domain",
+			},
+		},
+	}
+}
+
+func resourceDomainCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfClient := meta.(*SddcManagerClient)
+	apiClient := vcfClient.ApiClient
+
+	domainCreationSpec, err := createDomainCreationSpec(data)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	validateDomainSpec := domains.NewValidateDomainsOperationsParamsWithContext(ctx)
+	validateDomainSpec.DomainCreationSpec = domainCreationSpec
+
+	validateResponse, err := apiClient.Domains.ValidateDomainsOperations(validateDomainSpec)
+	if err != nil {
+		return validation_utils.ConvertVcfErrorToDiag(err)
+	}
+	if validation_utils.HasValidationFailed(validateResponse.Payload) {
+		return validation_utils.ConvertValidationResultToDiag(validateResponse.Payload)
+	}
+
+	domainCreationParams := domains.NewCreateDomainParamsWithContext(ctx)
+	domainCreationParams.DomainCreationSpec = domainCreationSpec
+
+	_, accepted, err := apiClient.Domains.CreateDomain(domainCreationParams)
+	if err != nil {
+		return validation_utils.ConvertVcfErrorToDiag(err)
+	}
+	taskId := accepted.Payload.ID
+	err = vcfClient.WaitForTaskComplete(taskId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	domainId, err := vcfClient.GetResourceIdAssociatedWithTask(taskId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(domainId)
+
+	return resourceDomainRead(ctx, data, meta)
+}
+
+func resourceDomainRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfClient := meta.(*SddcManagerClient)
+	apiClient := vcfClient.ApiClient
+
+	getDomainParams := domains.NewGetDomainParamsWithContext(ctx)
+	getDomainParams.ID = data.Id()
+	domainResult, err := apiClient.Domains.GetDomain(getDomainParams)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	domain := domainResult.Payload
+
+	_ = data.Set("name", domain.Name)
+	_ = data.Set("status", domain.Status)
+	_ = data.Set("type", domain.Type)
+	_ = data.Set("sso_id", domain.SSOID)
+	_ = data.Set("sso_name", domain.SSOName)
+	_ = data.Set("is_management_sso_domain", domain.IsManagementSSODomain)
+	if len(domain.VCENTERS) < 1 {
+		return diag.FromErr(fmt.Errorf("no vCenters found for domain %q", data.Id()))
+	}
+	_ = data.Set("vcenter_id", domain.VCENTERS[0].ID)
+	_ = data.Set("vcenter_fqdn", domain.VCENTERS[0].Fqdn)
+
+	err = readAndSetClustersData(domain.Clusters, data, apiClient)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	totalCpuCapacity := strconv.FormatFloat(domain.Capacity.CPU.Total.Value,
```

Comment:
See my comment earlier ... these kind of attributes are not required for automation. So, we could skip adding them to TF.

Reply:
Agree. Not every element value is required to be computed and stored in the state.